src/components/badge/notification_badge/badge_notification.styles.ts

```diff
@@ -57,6 +57,11 @@ export const euiNotificationBadgeStyles = (euiThemeContext: UseEuiTheme) => {
     accent: css`
       ${getBadgeColors(euiTheme.colors.accentText, euiThemeContext)}
     `,
+    success: css`
+      background-color: ${euiButtonFillColor(euiThemeContext, 'success')
+        .backgroundColor};
+      color: ${euiTheme.colors.ink};
+    `,
```

Comment:
Is there any reason why we can't use `getBadgeColors` here instead? I'm not a huge fan of the text color always being `#000` as opposed to responding dynamically to dark/light mode.

```suggestion
    success: css`
      ${getBadgeColors(euiTheme.colors.success, euiThemeContext)}
    `,
```

Reply:
This ended up being a fairly large refactor:
- https://github.com/elastic/eui/pull/6864/commits/307069615688285f08d2093f674aa30b7811f835
- https://github.com/elastic/eui/pull/6864/commits/d136d623223397b30347acb11bdb95bbc814884d
Badges now have a single utility/file for color logic, and the architecture with which they share colors now now more closely matches how EuiButton shares colors.

This change also matches the `hollow` appearance of EuiBadge and EuiBetaBadge, which were previously using slightly different text/border colors.